Account/views.py

```diff
@@ -91,36 +91,31 @@ def profile(request, username):
             if request.user == followed.user:
                 auth_follow = True
 
-    
     UserTweets = Tweet.objects.order_by('-pub_date').filter(tweet_creator=profile_user.pk)
 
-
     AllUsers = User.objects.all()
+    #If user currently authenticated...remove from QuerySet since we do not want their own profile displayed in 'WhoToFollow'
+    if request.user.is_authenticated:
+        AllUsers.exclude(username=request.user.username)
+    
     rand_three = []
-    for i in range (3):
+    #Keep looping until we either gather 3 new users to follow or run out of options with who to follow (Since the user follows everyone)
+    while len(rand_three) < 3 and len(AllUsers) > 0:
+        #Grab random user
         temp = random.choice(AllUsers)
-        while temp in rand_three or temp.username == request.user.username:
-            temp = random.choice(AllUsers)
+        #If we have an authenticated user currently logged in...check to see if we already follow this temp user
+        if request.user.is_authenticated and Follow.objects.filter(user=request.user, following=temp):
+            #Remove temp usery from Queryset if we already do
+            AllUsers = AllUsers.exclude(username=temp.username)
+            continue
+        #If temp is a correct user that we can follow..remove it from QuerySet and add it to the rand_three list
+        AllUsers = AllUsers.exclude(username=temp.username)
         rand_three.append(temp)
 
 
-    list_follow = [False, False, False]
-    if request.user.is_authenticated:
-
-
-        #Check to see if the authenticated user is already following the user
-        for i, listed_user in enumerate(rand_three, start=0):
-            temp_followed = Follow.objects.filter(following=listed_user)
-            for j in temp_followed:
-                if request.user == j.user:
-                    list_follow[i] = True
-
-    userlist = zip(rand_three, list_follow)
-
-
-    context = {'validSession':False, 'username':request.user.username, 'userdic':AllUsers, 'userlist':userlist, 
```

Comment:
Simplified the approach for grabbing the three accounts to follow.

Now we grab all the users to start, and as we add a user to the rand_three dictionary we remove it from the AllUsers list (QuerySet). If a given user is found that we already follow we go ahead and remove it from AllUsers. In the end we keep looping till we either find three users that we have not already followed yet, or we ran out of users that are viable for an account to follow.

This approach also removes some complexity.